### PR TITLE
Switch  flat_map tests to use TF combinations

### PR DESCRIPTION
**Pull request by feihugis**
This PR switches `flat_map` tests to use TF combination.

### CODE REVIEW

**Comment by feihugis** [comment]
@jsimsa Just in case you missed this PR, this is a gentle reminder to help review this PR when you get a chance.

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

File: tensorflow/python/data/kernel_tests/interleave_test.py
Comment (feihugis) [inline tensorflow/python/data/kernel_tests/interleave_test.py]: Change the original `input_values=[np.int64([4, 5, 6])]` to `input_values=[np.int64([4, 5, 6, 7])]`. Otherwise, there will be an error: 
```
absl.testing.parameterized.DuplicateTestNameError: Duplicate parameterized test name in InterleaveTest: generated test name 'testInterleaveDataset_test_mode_eager_tfapiversion_1_blocklength_3_cyclelength_1_inputvalues_456_numparallelcalls_None' (generated from 'testInterleaveDataset_test_mode_eager_tfapiversion_1_blocklength_3_cyclelength_1_inputvalues_456_numparallelcalls_None') already exists. Consider using named_parameters() to give your tests unique names and/or renaming the conflicting test method.
```
The cause is that when generating the test name, `dataset_ops.AUTOTUNE` will be converted to `1` instead of `-1`, which conflicts with the first combination (line 132 - 134)

**Comment by feihugis** [comment]
@jsimsa More tests are switched to use TF combination.

**Review by jsimsa — DISMISSED** [review]

**Review by jsimsa — DISMISSED** [review]

**Comment by feihugis** [comment]
The internal checks failed. Could you please help check the logs and paste them here?

**Comment by jsimsa** [comment]
```
======================================================================
FAIL: testSkipEagerNestedStructure_test_mode_eager_tfapiversion_1 (__main__.FromTensorsTest)
testSkipEagerNestedStructure_test_mode_eager_tfapiversion_1 (__main__.FromTensorsTest)
testSkipEagerNestedStructure_test_mode_eager_tfapiversion_1(mode='eager', tf_api_version=1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/work/bbfe7e7aa04b186c1d68ff6665bd9049dc07/google3/runfiles/google3/third_party/py/absl/third_party/unittest3_backport/case.py", line 37, in testPartExecutor
    yield
  File "/build/work/bbfe7e7aa04b186c1d68ff6665bd9049dc07/google3/runfiles/google3/third_party/py/absl/third_party/unittest3_backport/case.py", line 162, in run
    testMethod()
  File "/build/work/bbfe7e7aa04b186c1d68ff6665bd9049dc07/google3/runfiles/google3/third_party/py/absl/testing/parameterized.py", line 263, in bound_param_test
    test_method(self, **testcase_params)
  File "/build/work/bbfe7e7aa04b186c1d68ff6665bd9049dc07/google3/runfiles/google3/third_party/tensorflow/python/framework/test_combinations.py", line 310, in decorated
    execute_test_method()
  File "/build/work/bbfe7e7aa04b186c1d68ff6665bd9049dc07/google3/runfiles/google3/third_party/tensorflow/python/framework/test_combinations.py", line 298, in execute_test_method
    test_method(**kwargs_to_pass)
  File "/build/work/bbfe7e7aa04b186c1d68ff6665bd9049dc07/google3/runfiles/google3/third_party/tensorflow/python/data/kernel_tests/from_tensors_test.py", line 248, in testSkipEagerNestedStructure
    self.assertEqual([None, 3], w.shape.as_list())
AssertionError: Lists differ: [None, 3] != [1, 3]

First differing element 0:
None
1

- [None, 3]
+ [1, 3]
```

**Comment by feihugis** [comment]
@jsimsa Thanks for checking the internal failures! The failures are caused by the difference of output shapes in the graph and eager mode. The issues are addressed [here](https://github.com/tensorflow/tensorflow/pull/31380/commits/f6619348950c206667ff60e491dbe47a74c6c16c). Could you please have another look?

BTW, do we need to disable `FromTensorsTest` on mac as indicated [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/data/kernel_tests/BUILD#L245). I can run it on my mac locally.

**Comment by jsimsa** [comment]
Feel free to remove the "nomac" exclusion if the tests pass for you, but let's do it in a separate CL.

**Review by jsimsa — DISMISSED** [review]

**Comment by feihugis** [comment]
> Feel free to remove the "nomac" exclusion if the tests pass for you, but let's do it in a separate CL.

Sure, will do it once this PR gets merged.

**Comment by feihugis** [comment]
@jsimsa Sorry that I made a mistake in the last commit, which causes the method name conflicts. This commit(https://github.com/tensorflow/tensorflow/pull/31380/commits/1a6763540474873b5864d24430e26293dd6df360) changes the method name from `def testNestedStructure(...)` to `testNestedStructureV2(...)` to avoid the conflict with [here](https://github.com/tensorflow/tensorflow/pull/31380/files#diff-ee9cf39bad94aaf7df788842a74ce795L137).

File: tensorflow/python/data/kernel_tests/from_tensors_test.py
Comment (jsimsa) [inline tensorflow/python/data/kernel_tests/from_tensors_test.py]: This test does not logically belong to `from_tensors_test`. Move it to `dataset_test.py` and call it `testLegacyStructureAPI`.

File: tensorflow/python/data/kernel_tests/from_tensors_test.py
Comment (jsimsa) [inline tensorflow/python/data/kernel_tests/from_tensors_test.py]: rename this back to `testNestedStructure`

**Comment by feihugis** [comment]
@jsimsa Thanks for your suggestion. The last commit has been rewritten as you suggested. Please have a look at the change (https://github.com/tensorflow/tensorflow/pull/31380/commits/98aa8bbca941a7b7544ba4ae073e012226cedc4a).

**Review by jsimsa — APPROVED** [review]

**Comment by feihugis** [comment]
This PR has been pending for a while. Are there anything else I need to do?

**Comment by feihugis** [comment]
@jsimsa @rthadur This PR seems to be blocked. Could you please help check? Let me know if there is anything I can do.

**Comment by rthadur** [comment]
@feihugis sorry for the delay , let me try to pull this again , thank you for your patience.

**Comment by feihugis** [comment]
> @feihugis sorry for the delay , let me try to pull this again , thank you for your patience.

@rthadur No worries! Thanks for your help!

**Comment by rthadur** [comment]
This has been merged internally , waiting for auto-merge to happen, thank you

**Comment by feihugis** [comment]
Thank you, @rthadur!

**Pull request merged**
